OBSDOCS-797 - Short term authentication for cloud providers (#73846)
openshift/openshift-docs: modules/logging-identity-federation.adoc

// Module included in the following assemblies:
//

> // * list of assemblies where this module is included

Reviewer: ```suggestion

```

> // logging/log_storage/installing-log-storage.adoc

Reviewer: ```suggestion
// * logging/log_storage/installing-log-storage.adoc
```
Super-picky comment!

:_mod-docs-content-type: CONCEPT

> [id="<logging-identity-federation>_{context}"]

Reviewer: ```suggestion
[id="logging-identity-federation_{context}"]
```

> = Identity Federation

Reviewer: Should this be the following or is Identity  Federation a proper noun (which might need fixing throughout the docs)??
```suggestion
= Identity federation
```

> With {logging} 5.9 on {product-title} 4.14 and higher, if you use the {product-title} web console to install, STS clusters are automatically detected. You will be prompted to create roles and supply the data required for the Loki Operator to create a `CredentialsRequest` object, which will then populate a secret.

Reviewer: > if you use the {product-title} web console to install
What are we installing? How about:

```suggestion
If you use the {product-title} web console to install {logging} 5.9 on {product-title} 4.14 and higher, STS clusters are automatically detected. You will be prompted to create roles and supply the data required for the Loki Operator to create a `CredentialsRequest` object, which will then populate a secret.
```

> With {logging} 5.9 on {product-title} 4.14 and higher, if you use the {product-title} web console to install, STS clusters are automatically detected. You will be prompted to create roles and supply the data required for the Loki Operator to create a `CredentialsRequest` object, which will then populate a secret.

Reviewer: > if you use the {product-title} web console to install
What are we installing? How about:

```suggestion
For clusters using {logging} 5.9 on {product-title} 4.14 and higher, note the following about installing the Loki Operator:
* If you use the {product-title} web console to install the Operator, STS clusters are automatically detected. You will be prompted to create roles and supply the data required for the Loki Operator to create a `CredentialsRequest` object, which will then populate a secret.
* If you use the CLI to install the Operator, you must manually create a subscription object using the appropriate template for your storage provider.
```

> For installations using the CLI, create a subscription object using the appropriate template for your storage provider.

Reviewer: However you resolve my previous comment:

```suggestion
For installations using the the OpenShift CLI (`oc`)., create a subscription object using the appropriate template for your storage provider.
```

// Module included in the following assemblies:
// * logging/log_storage/installing-log-storage.adoc

:_mod-docs-content-type: PROCEDURE
[id="logging-identity-federation_{context}"]
= Workload identity federation

> Workload identity federation enables authentication to cloud based log stores using short lived tokens.

Reviewer: ```suggestion
Workload identity federation enables authentication to cloud-based log stores using short-lived tokens.
```

.Prerequisites

> * {product-title} 4.14 and higher

Author: Disagree. This is being introduced with the release of 5.9, which is supported on 4.13, but this functionality is not supported on 4.13.

> * {product-title} 4.14 and higher

Reviewer 2: Yes, but it looks like you only intend to CP this pull request to 4.14+, so it shouldn't matter since the docs will only show up on 4.14+?

> * {logging-uc} 5.9 and higher

Author: I somehow always forget this one, thanks. :)

.Procedure
* If you use the {product-title} web console to install the {loki-op}, STS clusters are automatically detected. You will be prompted to create roles and supply the data required for the {loki-op} to create a `CredentialsRequest` object, which will then populate a secret.

* If you use the OpenShift CLI (`oc`) to install the {loki-op}, you must manually create a subscription object using the appropriate template for your storage provider, as shown in the following examples. This authentication strategy is only supported for the storage providers indicated.

.Azure sample subscription
[source,yaml]
----
apiVersion: operators.coreos.com/v1alpha1
kind: Subscription
metadata:
  name: loki-operator
  namespace: openshift-operators-redhat
spec:
  channel: "stable-5.9"
  installPlanApproval: Manual
  name: loki-operator

>   source: qe-app-registry

Reviewer 3: the source should be `redhat-operators`

  sourceNamespace: openshift-marketplace
  config:
    env:

>       - name: CLIENT_ID

Reviewer 3: These should be `CLIENTID`, `TENANTID` and `SUBSCRIPTIONID` respectively

        value: <your_client_id>
      - name: TENANT_ID
        value: <your_tenant_id>
      - name: SUBSCRIPTION_ID
        value: <your_subscription_id>
      - name: REGION
        value: <your_region>
----

.AWS sample subscription
[source,yaml]
----
apiVersion: operators.coreos.com/v1alpha1
kind: Subscription
metadata:
  name: loki-operator
  namespace: openshift-operators-redhat
spec:
  channel: "stable-5.9"
  installPlanApproval: Manual
  name: loki-operator

>   source: qe-app-registry

Reviewer 3: the source should be 'redhat-operators'

  sourceNamespace: openshift-marketplace
  config:
    env:

>     - name: ROLE_ARN

Reviewer 3: this should be name: ROLEARN

>       value: "<role_ARN>"

Reviewer 4: The other example does not have quotes around the placeholder:

```suggestion
      value: <role_ARN>
```
